src/coil_read_write.cpp

```
@@ -0,0 +1,182 @@
+      request->leaf_id,
+      fc,
+      (uint8_t)((request->addr & 0xFF00) >> 8),   // high
+      (uint8_t)(request->addr & 0xFF),            // low
+      (uint8_t)((request->count & 0xFF00) >> 8),  // high
+      (uint8_t)(request->count & 0xFF),           // low
+      request_size,
+  };
+
+    // start filling out
+    for (size_t i = 0; i < request_size; ++i) {
+      data[7 + i ] = values_[i]; // high byte
```

cppcheck (2.10-2)

[error] Out of bounds access in expression 'values_[i]' because 'values_' is empty and 'i' may be non-zero.

(containerOutOfBounds)